engine/src/main/java/io/zeebe/engine/processor/workflow/handlers/element/ElementTerminatedHandler.java

```diff
@@ -44,15 +46,43 @@ protected boolean handleState(BpmnStepContext<T> context) {
                 flowScopeInstance.getState(), WorkflowInstanceIntent.ELEMENT_TERMINATED);
 
     incidentResolver.resolveIncidents(context);
+
     if (isScopeTerminating && isLastActiveExecutionPathInScope(context)) {
       context
           .getOutput()
           .appendFollowUpEvent(
               flowScopeInstance.getKey(),
               WorkflowInstanceIntent.ELEMENT_TERMINATED,
               flowScopeInstance.getValue());
+    } else if (wasInterrupted(context, flowScopeInstance)) {
+      publishInterruptingEventSubproc(context, flowScopeInstance);
     }
 
     return super.handleState(context);
   }
+
+  private void publishInterruptingEventSubproc(
```

Comment:
It's a bit different. It filters the records based on interrupting key, only publishes one and doesn't spawn tokens. I can refactor the `context.getOutput().appendFollowUpEvent(...)` to a method but it didn't seem significant

Reply:
Do we need to filter the deferred records? Do you have a case in mind when this could lead to an invalid situation?

Reply:
I'm wondering why we spawn the token for the event subprocess before and not like for the boundary event.

Reply:
My thinking was that we could have a process start event triggering and an interrupting even subprocess, do you think that we should publish everything regardless? I don't know about the boundary event but the idea with spawning the token before here was so that the scope isn't terminated when the other children are finished.

Reply:
If we clean up correctly then it should not trigger any event afterward. But yes, let's be safe with the check.

We already check if the scope was interrupted in the `ElementTerminatedHandler`. So, I guess that we don't need to spawn the token before. It works like it is now but it confuses me that it is different from the boundary event behavior. I would like to have only one way to do this :sweat_smile: